### ISSUE TITLE
Speed up library load time for CLI operation

### ISSUE THREAD

**Issue by boydgreenfield** [onecodex/onecodex#248]
Currently, we load a bunch of 3rd party libraries _when the user installs with `pip install onecodex[all]`_, which can add significant interpreter startup overhead to the CLI. We should profile and reduce this overhead where possible.

**Comment by boydgreenfield** [comment]
See this article for some profiling advice in Python 3.7: https://dev.to/methane/how-to-speed-up-python-application-startup-time-nkf

**Comment by boydgreenfield** [comment]
With Python 3.7 installed: `python -X importtime -c 'import onecodex' 2> onecodex-imports`

**Comment by jairideout** [comment]
Hi there! I'm interested in working on this issue if someone isn't already tackling it. Is it okay if I submit a pull request?

**Comment by boydgreenfield** [comment]
@jairideout Absolutely! Though FYI we did make a dent in this ~9 months ago and maybe should have closed this one (see https://github.com/onecodex/onecodex/blob/master/tests/test_speed.py). But if you can make it a bit faster (or if we've had regressions 😬), 100% that'd be fantastic.

**Comment by jairideout** [comment]
Great, I'll take a look!
